src/todo_file/src/lib.rs

```diff
@@ -391,14 +394,22 @@ impl TodoFile {
 	#[inline]
 	pub fn undo(&mut self) -> Option<(usize, usize)> {
 		self.version.increment();
-		self.history.undo(&mut self.lines)
+		if let Some((operation, start, end)) = self.history.undo(&mut self.lines) {
+			return if operation == Operation::Load {
+				None
```

Comment:
@forivall , for #873 , I think you should be able to call `detect_state` here, to reset the state back to the initial state.

Thoughts?

Reply:
Yup, that seems correct